smdebug/tensorflow/keras.py

```diff
@@ -757,6 +757,13 @@ def _save_layer_values(self, logs):
         for layer_name, layer_input, layer_output in logs:
             # Cast layer_name to str since it can also be of type bytes
             # when run with mirrored strategy
+            if isinstance(layer_name, tf.Tensor):
```

Comment:
isn't this problem there with other tensors that are saved from logs? gradients, metrics etc?

Reply:
Gradients already has this logic.
The problem happens from passing strings from the tf layer; so it will not happen with metrics.